pyfarm/jobtypes/core/jobtype.py

```
@@ -890,6 +890,96 @@ def set_task_progress(self, task, progress):
+            try:
+                response = yield post_direct(url, data=data)
+                response_data = yield treq.json_content(response)
+                if response.code == OK:
+                    logger.info("Set time_started of task %s to now on master",
+                                task ["id"])
+                    updated = True
+                    returnValue(None)
+
+                elif response.code >= INTERNAL_SERVER_ERROR:
+                    delay = http_retry_delay()
+                    logger.warning(
+                        "Could not post start time for task %s to the "
+                        "master server, internal server error: %s.  Retrying "
+                        "in %s seconds.", task["id"], response_data, delay)
+
+                    deferred = Deferred()
+                    reactor.callLater(delay, deferred.callback, None)
+                    yield deferred
+
+                elif response.code == NOT_FOUND:
+                    message = ("Got 404 NOT FOUND error on setting start time "
+                               "for task %s" % task["id"])
+                    logger.error(message)
+                    raise Exception(message)
+
+                elif response.code >= BAD_REQUEST:
+                    message = (
+                        "Failed to set start time for task %s on the "
+                        "master, bad request: %s. Server.  This request will "
+                        "not be retried." % (task["id"], response_data))
+                    logger.error(message)
+                    raise Exception(message)
+
+                else:
+                    message = (
+                        "Unhandled error when setting start time for task "
+                        "%s to the master: %s (code: %s).  This request will "
+                        "not be retried." % (response_data, response.code))
+                    logger.error(message)
+                    raise Exception(message)
```

Similar note as above, this is a generic catch-all handler, I cannot think of a better fitting exception name here.